src/pixelator/pixeldataset/utils.py

```diff
@@ -217,6 +219,76 @@ def write_anndata(adata: AnnData, filename: PathType) -> None:
     adata.write(filename=filename, compression="gzip")
 
 
+def _compute_sub_communities(
+    component_edgelist: pd.DataFrame, n_edges_reconnect: int | None = None
+):
+    component_edgelist = (
+        component_edgelist.groupby(["upia", "upib"], observed=True)["count"]
+        .count()
+        .reset_index()
+        .sort_values(["upia", "upib"])
+    )
+    edgelist_tuple = list(
+        map(tuple, np.array(component_edgelist[["upia", "upib", "count"]]))
+    )
+    component_communities_dict = leiden(
+        edgelist_tuple,
+        resolution=0.5 * LEIDEN_RESOLUTION,
```

Comment:
Maybe we can include a comment here on the decision we took in the ticket about doing 1/2 of the resolution

Reply:
Maybe we can include it within a single-line docstring on what this function is going to achieve.